### DISCUSSION TITLE
how to get the label payload of labeld/unlabled event

### DISCUSSION THREAD

**Discussion by sixwaaaay** [octokit/webhooks.net#406] [Q&A] [answered]
labeld/unlabeld event belongs to issue event, but there is no label payload which triggered the event in the signature of handler `ProcessIssuesWebhookAsync(WebhookHeaders headers, IssuesEvent issuesEvent,IssuesAction action)` .

**Comment by wolfy1339** [comment]
The label that was added or removed is in the `label` property of the webhook payload.

Which in this case seems like the `issuesEvent` parameter

**Reply by sixwaaaay** [reply]
IssuesEvent:![image](https://github.com/octokit/webhooks.net/assets/56881699/eb9af61b-7562-431a-832d-47bcc1251a6c)
![image](https://github.com/octokit/webhooks.net/assets/56881699/3466e3d5-dda2-4996-a16c-e5c0c47c5846)
AdditionalProperties is null, 
![image](https://github.com/octokit/webhooks.net/assets/56881699/b159b935-1056-4542-97fb-96ddb915a095)
When an unlabeld event causes an issue to no longer have a label, nothing happens to this code



proxy capture: 
![image](https://github.com/octokit/webhooks.net/assets/56881699/f1733a8c-cd10-4c5d-aba6-697f89e31871)

![image](https://github.com/octokit/webhooks.net/assets/56881699/9fed3308-97b1-4b04-8da4-264f85ed1e39)

![image](https://github.com/octokit/webhooks.net/assets/56881699/b6b761ec-d113-47b6-a073-57c8d8c1d3de)

**Comment by sixwaaaay** [comment, marked as answer]
![image](https://github.com/octokit/webhooks.net/assets/56881699/59d2c693-42a8-44cd-ab41-7fd514b0164d)
But maybe there's an easier way.

**Reply by JamieMagee** [reply]
Does this work for you?

```csharp
protected override Task ProcessIssuesWebhookAsync(WebhookHeaders headers, IssuesEvent issuesEvent, IssuesAction action)
    {
        var label = issuesEvent switch
        {
            IssuesLabeledEvent labeled => labeled.Label?.Name,
            IssuesUnlabeledEvent unlabeled => unlabeled.Label?.Name,
            _ => null
        };

        return Task.CompletedTask;
    }
```

**Reply by sixwaaaay** [reply]
May I ask if webhooks.net has plans to support AOT?

**Reply by JamieMagee** [reply]
What errors are you getting when compiling with AOT? Can you create an issue please?

**Reply by sixwaaaay** [reply]
no errors when compiling with AOT, only warning when `dotnet publish -c Release`:
```
octokit.webhooks\2.0.6\lib\net6.0\Octokit.Webhooks.dll : warning IL2104: Assembly 'Octokit.Webhooks' prod
uced trim warnings. For more information see https://aka.ms/dotnet-illink/libraries [xxxx\Bot.csproj]
octokit.webhooks\2.0.6\lib\net6.0\Octokit.Webhooks.dll : warning IL3053: Assembly 'Octokit.Webhooks' prod
uced AOT analysis warnings. [xxxx\Bot.csproj]
octokit\9.0.0\lib\netstandard2.0\Octokit.dll : warning IL2104: Assembly 'Octokit' produced trim warnings.
 For more information see https://aka.ms/dotnet-illink/libraries [xxxx\Bot.csproj]
octokit\9.0.0\lib\netstandard2.0\Octokit.dll : warning IL3053: Assembly 'Octokit' produced AOT analysis w
arnings. [xxxx\Bot.csproj]
```

Exception throws when deserializing the received webhook payload.
```c#
fail: Octokit.Webhooks.WebhookEventProcessor[3]
      Exception processing GitHub event.
      System.InvalidOperationException: Reflection-based serialization has been disabled for this application. Either use the source generator APIs or explicitly configure the 'JsonSerializerOptions.TypeInfoResolver' property.
         at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_JsonSerializerIsReflectionDisabled() + 0x2b
         at System.Text.Json.JsonSerializerOptions.ConfigureForJsonSerializer() + 0x31
         at System.Text.Json.JsonSerializer.GetTypeInfo(JsonSerializerOptions, Type) + 0x2a
         at System.Text.Json.JsonSerializer.GetTypeInfo[T](JsonSerializerOptions) + 0x25
         at System.Text.Json.JsonSerializer.Deserialize[TValue](String, JsonSerializerOptions) + 0x28
         at Octokit.Webhooks.WebhookEventProcessor.DeserializeWebhookEvent(WebhookHeaders, String) + 0xb55
         at Octokit.Webhooks.WebhookEventProcessor.ProcessWebhookAsync(IDictionary`2, String) + 0x36
         at Octokit.Webhooks.AspNetCore.GitHubWebhookExtensions.<>c__DisplayClass0_0.<<MapGitHubWebhooks>b__0>d.MoveNext() + 0x262

```
